packages/energy-api-influxdb/src/reads/reads.service.ts

```diff
@@ -100,11 +100,12 @@ export class ReadsService implements OnModuleInit {
   public async findLatestRead(
     meterId: string,
     startDuration?: AllowedDurationType,
-  ): Promise<ReadDTO[]> {
+  ): Promise<ReadDTO> {
     try {
       const query = this.findLatestReadByMeterQuery(meterId, startDuration);
 
-      return this.execute(query);
+      const reads = await this.execute(query);
+      return reads[0];
```

Comment:
We shouldn't try to access the index 0 element before checking if the element exists.

```ts
if (reads.length === 0) {
    throw new NotFoundException(`Unable to get the latest reading. There are no readings yet for meter ${meterId}`);
} else {
    return reads[0];
}
```

Reply:
yep